docs/docs/building-modules/09-module-interfaces-continued.md

```diff
@@ -0,0 +1,158 @@
+
+### Commands and Queries
+
+The `autocli` package generates CLI commands and flags for each method defined in your gRPC service. By default, it generates commands for each RPC method that does not return a stream of messages. The commands are named based on the name of the service method.
+
+For example, given the following protobuf definition for a service:
+```
+service MyService {
+  rpc MyMethod(MyRequest) returns (MyResponse) {}
+}
+
+```
+
+`autocli` will generate a command named `my-method` for the `MyMethod` method. The command will have flags for each field in the `MyRequest` message.
+
+If you want to customise the behavior of a command, you can define a custom command by implementing the `autocli.Command` interface. You can then register the command with the `autocli.Builder` instance for your application.
+
+Similarly, you can define a custom query by implementing the `autocli.Query` interface. You can then register the query with the `autocli.Builder` instance for your application.
+
+To add a custom command or query, you can use the `Builder.AddCustomCommand` or `Builder.AddCustomQuery` methods, respectively. These methods take a `cobra.Command` or `cobra.Command` instance, respectively, which can be used to define the behavior of the command or query.
+
+### Advanced Usage
+
+#### Specifying Subcommands
+
+By default, `autocli` generates a command for each method in your gRPC service. However, you can specify subcommands to group related commands together. To specify subcommands, you can use the `autocliv1.ServiceCommandDescriptor` struct.
+
+For example, suppose you have a service with two methods: `GetTest` and `GetTest2`. You can group these methods together under a `get` subcommand using the following code
+
+```
+autoCliOpts := &autocliv1.ModuleOptions{
+    Tx: &autocliv1.ServiceCommandDescriptor{
+        Service: myServiceDesc.ServiceName,
+        SubCommands: map[string]*autocliv1.ServiceCommandDescriptor{
+            "get": {
+                Service: myServiceDesc.ServiceName,
+                SubCommands: map[string]*autocliv1.ServiceCommandDescriptor{
+                    "test": {Service: myServiceDesc.ServiceName, Method: "GetTest"},
+                    "test2": {Service: myServiceDesc.ServiceName, Method: "GetTest2"},
+                },
+            },
+        },
+    },
+}
+
+
+```
+
+With this configuration, you can invoke the `GetTest` method by running `./app tx myservice get test`, and you can invoke the `GetTest2` method by running `./app tx myservice get test2`.
+
+#### Adding Custom Commands
+
+Although `autocli` can automatically generate CLI commands and flags based on your Protocol Buffer messages, you may want to add custom commands to your CLI application. To add a custom command, you can use the `AddCommand()` method provided by `cobra`. For instance, to add a custom `test` command to your CLI application, you can use the following code:
```

Comment:
```suggestion
Although `autocli` can automatically generate CLI commands and flags based on your protobuf messages, you may want to add custom commands to your CLI application. To add a custom command, you can use the `AddCommand()` method provided by `cobra`. For instance, to add a custom `test` command to your CLI application, you can use the following code:
```

Can we sometime replace `Protocol Buffer` by `protobuf`? imho it reads easier.
Moreover, maybe we have said enough that it can generate commands?

It would otherwise simplify this section by saying that autocli gives you a cobra command, so you are free to use the cobra API to add commands or customize the behavior.

Reply:
We could also remove this section completely too if you see fit. tbh I wasnt sure if we needed this section as much

Reply:
I think it good to briefly mention it is a cobra command, so it does not behave differently.
Removing it does make sense too, as you can anyway see what the function returns.

Reply:
I think lets remove then if it seems already a lil repetitive